larray_editor/editor.py

```diff
@@ -275,6 +279,29 @@ def get_value(self):
         # already been destroyed, due to the Qt.WA_DeleteOnClose attribute
         return self.data
 
+    def add_widget_with_state_to_restore(self, widget_name, widget):
+        if hasattr(widget, 'saveState'):
+            name = 'state/{}'.format(self.__class__.__name__)
```

Comment:
I am not entirely sure I like the use of self.__class__.__name__ as a part of settings keys because user settings will break if we ever change change the class names. Maybe adding a class level attribute for the settings key for that class would decouple that?